torch/nn/parallel/distributed.py

```diff
@@ -1344,6 +1344,11 @@ def forward(self, *inputs, **kwargs):
                 assert self.logger is not None
                 self.logger.set_runtime_stats_and_log()
                 self.num_iterations += 1
+                if self.num_iterations > 1 and any(
+                    p.grad is None for p in self.module.parameters()
+                ):
+                    self.reducer._point_grads_to_bucket()
```

Comment:
wondering whether it will be better to move to the end of forward, so that it is close to beginning of backward just in case:)